Missing `int64_t` declaration in rectangular_lsap.cpp

I was trying to install scipy for several hours in `python:3.7-alpine` image. Unfortunately it doesn't build for me. Thought it was a problem with dependencies, tried multiple configurations, but then checked the code that's actually failing and found an issue.

#### Steps to reproduce:

Create `dummy.dockerfile` with following content:

```
FROM python:3.7-alpine
RUN apk add --no-cache build-base openblas-dev lapack-dev
RUN pip install scipy==1.4.1
```

and run:

```
docker build -t dummy -f dummy.dockerfile .
```

#### Error message:

```
creating build/temp.linux-x86_64-3.7/scipy/optimize/rectangular_lsap
  compile options: '-I/tmp/pip-build-env-psdpbyuc/overlay/lib/python3.7/site-packages/numpy/core/include -c'
  g++: scipy/optimize/rectangular_lsap/rectangular_lsap.cpp
  Running from scipy source directory.
  /tmp/pip-build-env-psdpbyuc/overlay/lib/python3.7/site-packages/numpy/distutils/system_info.py:716: UserWarning: Specified path /usr/include/python3.7m is invalid.
    return self.get_paths(self.section, key)
  scipy/optimize/rectangular_lsap/rectangular_lsap.cpp:117:43: error: 'int64_t' has not been declared
    117 | solve(int nr, int nc, double* input_cost, int64_t* output_col4row)
        |                                           ^~~~~~~
  scipy/optimize/rectangular_lsap/rectangular_lsap.cpp:186:41: error: 'int64_t' has not been declared
    186 |                                         int64_t* col4row)
        |                                         ^~~~~~~
  scipy/optimize/rectangular_lsap/rectangular_lsap.cpp:117:43: error: 'int64_t' has not been declared
    117 | solve(int nr, int nc, double* input_cost, int64_t* output_col4row)
        |                                           ^~~~~~~
  scipy/optimize/rectangular_lsap/rectangular_lsap.cpp:186:41: error: 'int64_t' has not been declared
    186 |                                         int64_t* col4row)
        |                                         ^~~~~~~
  error: Command "g++ -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -DTHREAD_STACK_SIZE=0x100000 -fPIC -I/tmp/pip-build-env-psdpbyuc/overlay/lib/python3.7/site-packages/numpy/core/include -c scipy/optimize/rectangular_lsap/rectangular_lsap.cpp -o build/temp.linux-x86_64-3.7/scipy/optimize/rectangular_lsap/rectangular_lsap.o -MMD -MF build/temp.linux-x86_64-3.7/scipy/optimize/rectangular_lsap/rectangular_lsap.o.d" failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for scipy
```

#### Scipy/Numpy/Python version information:

```
scipy==1.4.1
```

#### My findings:

I was digging a bit through the code and found that `int64_t` declaration is truly missing in the [`rectangular_lsap.cpp`](https://github.com/scipy/scipy/blob/master/scipy/optimize/rectangular_lsap/rectangular_lsap.cpp) file. A line with

```
#include <cstdint>
```

should be added.

EDIT: @peterbell10 suggested that it may be better to

```
#include "rectangular_lsap.h"
```

--- a/THANKS.txt
+++ b/THANKS.txt
@@ -227,6 +227,7 @@ Leo P. Singer for bug fixes in scipy.optimize and scipy.interpolate and
 Domen Gorjup and Janko Slavič for continuous wavelet transform with complex
     wavelets fix.
 Søren Fuglede Jørgensen for improvements to scipy.sparse.csgraph
+Grzegorz Mrukwa for a bug fix in rectangular_lsap.cpp
 
 Institutions
 ------------

--- a/scipy/optimize/rectangular_lsap/rectangular_lsap.cpp
+++ b/scipy/optimize/rectangular_lsap/rectangular_lsap.cpp
@@ -42,7 +42,7 @@ Author: PM Larsen
 
 #include <algorithm>
 #include <cmath>
-#include <cstdint>
+#include "rectangular_lsap.h"
 #include <vector>
 
 static int

--- a/scipy/optimize/rectangular_lsap/rectangular_lsap.cpp
+++ b/scipy/optimize/rectangular_lsap/rectangular_lsap.cpp
@@ -42,6 +42,7 @@ Author: PM Larsen
 
 #include <algorithm>
 #include <cmath>
+#include <cstdint>
 #include <vector>
 
 static int

